Dashboards Versions: Add event to dashboards restore version

yaelleC opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

Adding a new event to the codebase, `grafana_dashboards_version_restore_clicked` with props:
- `version` : the version to restore 
- `index` : the index of the row the version was shown in (the highest the number, the lower on the table it was)
- `confirm` : true is the click was on the confirm button

This PR also adds a `grafana_dashboards_show_more_versions_clicked` event to the `Show more versions` button

**Why do we need this feature?**

To make a decision on history to save in app platform 


**Who is this feature for?**

App platform (@stephaniehingtgen / @frossano-grafana) for starters, we can remove the event after the decision is made - or keep it if dashboards PM (@thanos-karachalios ) considers there could be UI iterations based on the data.

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #98856 

**Special notes for your reviewer:**
This is the non-scenes PR,  a scenes PR should follow.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

yaelleC committed:
Add event to dashboards restore version

yaelleC committed:
moved changes to scenes instead

yaelleC committed:
remove last non-scene change

ivanortegaalba reviewed (approved):
Looks good to me! 💯

ivanortegaalba reviewed (commented):
I'd recommend to add some unit tests to avoid regressions if anybody changes the code in the future :)

axelavargas reviewed (commented):
Hey Yaelle 👋🏾, the change looks good, I think we will need to move it to the centralized interactions file https://github.com/grafana/grafana/blob/main/public/app/features/dashboard-scene/utils/interactions.ts

yaelleC commented:
> Hey Yaelle 👋🏾, the change looks good, I think we will need to move it to the centralized interactions file https://github.com/grafana/grafana/blob/main/public/app/features/dashboard-scene/utils/interactions.ts

@axelavargas  Amazing! That's the type of file I was looking for, I'll add them there! Much better 😉

yaelleC committed:
Move events to centralised file

yaelleC committed:
remove ? from properties copy pasta

yaelleC committed:
Add tests

yaelleC commented:
@axelavargas I moved the events to the centralised file, let me know if it looks okay. 
FYI: I added type definitions as I feel that's the huge benefit you get from having a central file: you get to decide what props should be sent with the event and avoid whole objects or wrong things sent by mistake (I almost sent the whole `version` json object but the fact that my new function is expecting a number stopped me 😛 )

@ivanortegaalba I added a couple of test files. It's meant to be a temporary event (unless Thanos think it's valuable outside of deciding how many versions to keep) so I'll remove the file when I remove the event 😊

axelavargas reviewed (approved):
great! code wise looks good to me 🙌🏾
